### ISSUE TITLE
Persistent wrong search “query”

### ISSUE THREAD

**Issue by neoOpus** [hoothin/SearchJumper#92]
**Describe the bug**
When using Microsoft Edge version 117 in conjunction with a userscript designed to switch search engines, a critical issue arises. The userscript fails to utilize the search engine specified in the search box, instead opting for an alternate search engine. Even after attempting to refresh the page, the issue persists, necessitating manual rewriting or copy-pasting of the desired search query to use a different search engine. This bug undermines the intended convenience of the userscript, which is supposed to facilitate effortless transitions between search engines.

**To Reproduce**

Open Microsoft Edge
Install and activate the userscript designed for switching search engines.
Navigate to a web page where the userscript is intended to work.
Attempt to switch the search engine using the userscript.
Observe that the userscript does not use the search engine specified in the search box, but instead defaults to a different one.
Even after refreshing the page, the issue persists, requiring manual input or copy-pasting to use the desired search engine.

**Expected behavior**
The expected behavior is that when using the userscript in Microsoft Edge 117, it should seamlessly switch to the search engine specified in the search box without any manual intervention. Refreshing the page should not be necessary to achieve this functionality.

**Desktop (please complete the following information):**
 - OS: Windows 11 (latest insider)
 - Browser MS Edge (latest)
 - Version 117

**Comment by hoothin** [comment]
What do you mean about "search box"? SearchJumper won't change your default search engine specified in the search box.
Have you checked other addons or scripts on your platform?

**Comment by neoOpus** [comment]
I am referring to the search field in search engines, which holds the search text... It displays something, but when I switch to another search engine using SearchJumper, it doesn't use that, but instead uses another text that I searched in another tab... This issue has been present for a long time, but I initially attributed it to my faulty OS and Browser, which is why I didn't report it earlier. However, even with a fresh OS, I am still experiencing this problem. I hope you can reproduce and resolve it.

**Comment by hoothin** [comment]
Can you post your rule of that engine here?
The keywords will restore to cached keywords if script can't find one on current site. Then you should edit rule by yourself to extract
![image](https://github.com/hoothin/SearchJumper/assets/5361213/50e818ec-0345-44e2-977d-df17cc74254d)

**Comment by neoOpus** [comment]
Hi,

Currently, it may be due to the caching mechanism. This issue is not related to the search engine. When I perform a fresh search, everything works as expected. However, when I leave some search tabs open and come back to them later, this problem occurs. It performs strange queries that include not only the previously searched keyword but also other text from the URL, such as the Wikipedia skin.
I will provide more detailed steps on how to reproduce it if you are unable to do so on your own.


![image](https://github.com/hoothin/SearchJumper/assets/1388336/2b5da8c9-52a6-4b16-9289-66acd7a6c542)

**Comment by hoothin** [comment]
No, I can't reproduce, did you check the url just when you open the page?
A video or gif maybe helpful.

**Comment by neoOpus** [comment]
I will try to upload a video for this issue, I just have to fix my ShareX not being able to capture a video.
